.github/workflows/build-test.yaml

```
@@ -43,45 +42,24 @@ jobs:
         sudo snap install cmake --classic
         hash -r
         cmake --version
-        mkdir -p build
-        cd build
-        cmake ../
-        cmake --build . -- -j 6
-        echo "Running ./src/runtest"
-        ./src/runtest
-        valgrind --leak-check=full --show-leak-kinds=all --errors-for-leak-kinds=all  ./src/runtest
 
-    - name: Mac OS build C++ and test
+    - name: Prepare build system for macOS
       if: startsWith(matrix.os, 'macos')
       run: |
         ls -l
         export MACOSX_DEPLOYMENT_TARGET=10.14
-        mkdir -p build
-        ls -l build
-        cd build
-        cmake ../
-        cmake --build . -- -j 6
-        echo "Running ./src/runtest"
-        ./src/runtest
-
-    - name: Test pure python implementation
-      run: |
-        python python-impl/impl-test.py
-
-    - name: Install emsdk
-      uses: mymindstorm/setup-emsdk@v11
+        brew install autoconf automake gmp
 
-    - name: Test javascript bindings
+    - name: Build library using CMake
       run: |
-        emcc -v
-        sh emsdk_build.sh
-        sh js_test.sh
+        mkdir -p build && cd build
+        CC=${{ matrix.compiler.cc }} CXX=${{ matrix.compiler.cxx }} cmake .. -DARITH=${{ matrix.backend }}
```

this should fix tests (worked for me https://github.com/UdjinM6/bls-signatures/pull/1)
```suggestion
        CC=${{ matrix.compiler.cc }} CXX=${{ matrix.compiler.cxx }} cmake .. -DARITH=${{ matrix.backend }} -DBUILD_BLS_PYTHON_BINDINGS=0
```